test/units/parsing/test_splitter.py

```diff
@@ -173,3 +173,12 @@ def test_split_args_error(args):
 def test_parse_kv_error(args):
     with pytest.raises(AnsibleParserError):
         parse_kv(args)
+
+
+@pytest.mark.parametrize("args", (
+    'creates=something thisisextra',
+    'something thisisextra',
+))
+def test_parse_kv_remainder_error(args: str) -> None:
+    with pytest.raises(ValueError, match=r'^Invalid k=v detected: .*$'):
```

Comment:
It's probably not a bad idea to verify the string starts with the expected message, but it's pointless to check the end of the string, so this can be simplified:

```suggestion
    with pytest.raises(ValueError, match=r'^Invalid k=v detected: '):
```